dat/missions/neutral/adblocker.lua

```diff
@@ -0,0 +1,164 @@
+--[[
+<?xml version='1.0' encoding='utf8'?>
+<mission name="Adblocker">
+ <unique />
+ <chance>20</chance>
+ <location>Bar</location>
+ <faction>Dvaered</faction>
+ <faction>Empire</faction>
+ <faction>Frontier</faction>
+ <faction>Goddard</faction>
+ <faction>Independent</faction>
+ <faction>Sirius</faction>
+ <faction>Soromid</faction>
+ <faction>Za'lek</faction>
+</mission>
+--]]
+--[[
+    Adblocker
+    Author: Lemuria
+
+    Player has to kill a ship that's been spamming the local comms with
+    excessive advertisements.
+
+   I was bored and had nothing better to do, so I made this Naev mission,
+   my first.
+--]]
+
+local fmt = require "format"
+local neu = require 'common.neutral'
+local vn = require "vn"
+local vntk = require "vntk"
+local portrait = require "portrait"
+
+local mission = {
+   name = _("Adblocker"),
+   description = "Destroy a ship spamming the {sys} system with excessive advertising transmissions",
+   reward = 300e3,
+   npc = {
+      name = "Desperate captain",
+      img = "neutral/male7.webp",
+      description = _("You see a desperate looking captain.")
+   }
+}
+
+-- fp means "full path"
+mission.npc["img_fp"] = portrait.getFullPath(mission.npc.img)
+
+function create()
+   mem.current_system = system.cur()
+   mem.current_spob = spob.cur()
+   misn.setNPC(mission.npc.name, mission.npc.img, mission.npc.description)
+end
+
+function accept()
+   local accepted = false
+
+   vn.clear()
+   vn.scene()
+   local man = vn.newCharacter(mission.npc.name, { image = mission.npc.img_fp })
+   vn.transition()
+   man(fmt.f(_([["Look, I don't have much time! There's this annoying ship that's been spamming the local comms with tons of advertisements! I can't take it anymore! Please, you've got to stop it! I'll give you ${creds} if you stop it!"]])
+      , { creds = fmt.credits(mission.reward) }))
+   vn.menu {
+      { _([[Accept]]), "accept" },
+      { _([[Refuse]]), "refuse" },
+   }
+
+   vn.label("refuse")
+   vn.done()
+
+   vn.label("accept")
+   man(_([["Thank you! Again, just stop that thing! Do what you have to!"]]))
+   vn.func(function()
+      accepted = true
+   end)
+   vn.run()
+
+   if not accepted then return end
+
+   misn.accept()
+   misn.setTitle(mission.name)
+   misn.setDesc(fmt.f(_("A ship is currently spamming {sys} with tons of unwanted advertisements. A desperate captain has asked you to destroy, or disable it.")
+      , { sys = mem.current_system }))
+   misn.setReward(fmt.credits(mission.reward))
+   misn.osdCreate(mission.name, {
+      mission.name,
+      fmt.f(_(mission.description), { sys = mem.current_system })
```

Comment:
You should probably add the "Return to planet" line in the osd here, and then use `misn.osdActive(2)` to switch to it instead of recreating the osd.